rasa/core/processor.py

```diff
@@ -733,7 +731,7 @@ async def parse_message(
     def _parse_message_with_graph(
         self,
         message: UserMessage,
-        tracker: DialogueStateTracker,
+        tracker: Optional[DialogueStateTracker] = None,
```

Comment:
tracker might be None due to it being optional in _parse_message_with_graph - IMO we could create an empty tracker in the processor if there is none to avoid needing it to handle here. otherwise it is a bit odd because you have a tracker provider that could be none but also can't for other components.

otherwise your nlu component must handle None - currently have a hidden type mismatch there in `process`